extensions/kafka/src/main/java/io/deephaven/kafka/KafkaTools.java

```diff
@@ -127,19 +139,128 @@ public static Schema getAvroSchema(final String schemaServerUrl, final String re
                 throw new UncheckedDeephavenException("Got status code " + statusCode + " requesting " + request);
             }
             action = "extract json server response";
-            final HttpEntity entity = response.getEntity();
-            final Header encodingHeader = entity.getContentEncoding();
-            final Charset encoding = encodingHeader == null
-                    ? StandardCharsets.UTF_8
-                    : Charsets.toCharset(encodingHeader.getValue());
-            final String json = EntityUtils.toString(entity, encoding);
+            final String json = extractSchemaServerResponse(response);
             action = "parse schema server response: " + json;
             return new Schema.Parser().parse(json);
         } catch (Exception e) {
             throw new UncheckedDeephavenException("Exception while trying to " + action, e);
         }
     }
 
+    /**
+     * Push an Avro schema from a Confluent compatible Schema Server.
+     *
+     * @param schema An Avro schema
+     * @param schemaServerUrl The schema server URL
+     * @param resourceName The resource name that the schema will be known as in the schema server
+     * @return The version for the added resource as returned by schema server.
+     */
+    public static String putAvroSchema(
+            final Schema schema, final String schemaServerUrl, final String resourceName) {
+        String action = "setup http client";
+        try (final CloseableHttpClient client = HttpClients.custom().build()) {
+            final String requestStr =
+                    schemaServerUrl + "/subjects/" + resourceName + "/versions/";
+            final List<NameValuePair> urlParameters = new ArrayList<>();
+            final String jsonSchema = schema.toString().replaceAll("[\\n\\r]", "");
```

Comment:
It certainly feels yucky not depending on a JSON library to encode JSON here, and also not using a common escape thing like Apache string utils, but rather doing string manipulation by hand.